src/plugins/event-plugins/FetchPlugin.ts

```diff
@@ -85,11 +87,26 @@ export class FetchPlugin extends MonkeyPatched<Window, 'fetch'> {
         init: RequestInit | undefined,
         argsArray: IArguments
     ): XRayTraceEvent => {
+        let traceId;
+        let segmentId;
+
+        const headers = (input as Request).headers;
+        if (headers) {
+            const headerComponents = headers.get(X_AMZN_TRACE_ID)?.split(';');
+            if (headerComponents?.length === 3) {
+                traceId = headerComponents[0].split('Root=')[1];
+                segmentId = headerComponents[1].split('Parent=')[1];
+            }
+        }
```

Comment:
nit: Stick this in a helper function, and add unit tests. http-utils might be a good place.